src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -160,6 +240,15 @@ impl WorkUnitStore {
       .workunit_records
       .insert(span_id.clone(), WorkunitRecord::Started(started));
     inner.started_ids.push(span_id.clone());
+    let child = inner.graph.add_node(span_id.clone());
+    inner
+      .span_id_to_graph
+      .insert(span_id.clone(), child.clone());
+    if let Some(parent_id) = parent_id {
+      let parent = *inner.span_id_to_graph.get(&parent_id).unwrap();
```

Comment:
I can't really imagine why this wouldn't already exist, but it feels better to lose the connection between the nodes than to panic here.

Reply:
makes sense